oslo_cache/backends/redis.py

```diff
@@ -0,0 +1,45 @@
+from dogpile.cache.backends import redis as redis_backend
+from oslo_cache import core
+__all__ = [
+    'RedisSentinelTLSBackend'
+]
+class RedisSentinelTLSBackend(redis_backend.RedisSentinelBackend):
+    def __init__(self,arguments):
+        """
+        Wrapper class for dogpile.cache.redis_sentinel.
+        Currently only translates arguments for dogpile.cache.backends.RedisSentinelBackend
+        because user is unable to pass them using oslo config in it's current format
+        arguments:
+           sentinels: list of sentinels formated as sentinel1IP:port,sentinel2IP:port
+           ssl: turns on TLS
+           ssl-<keyfile,certfile,ca_certs>: TLS certificate, key and CA
+           username,password: login credentials 
+        """
+        # Reformat sentinels into (ip <string>, port <int>) as required by dogpile.cahce.redis_sentinel
+        sentinels=[tuple([ int(spl) if i==1 else spl for i,spl in enumerate(ip_port.split(":"))]) for ip_port in arguments['sentinels'].split(',')]
```

Comment:
There are a few points here:
* https://www.flake8rules.com/rules/E225.html
* https://www.flake8rules.com/rules/E201.html
* https://www.flake8rules.com/rules/E231.html
* https://www.flake8rules.com/rules/E501.html

As per https://peps.python.org/pep-0020/ I would not try to put everything on one line just because it can be on one line, turning this into something like this:
```suggestion
        sentinels = [
            tuple(
                [
                    int(spl) if i == 1 else spl
                    for i, spl in enumerate(ip_port.split(":"))
                ]
            )
            for ip_port in arguments["sentinels"].split(",")
        ]
```
Make this much much more readable and understandable.

The approach you choose works, but I think something like this may be a bit nicer:
```python
        sentinels = [
            (ip, int(port))
            for ip, port in [
                ip_port.split(":") for ip_port in arguments["sentinels"].split(",")
            ]
	]
```

There is also one possible bug: support for IPv6, which uses `:` as a separator.

This can be fixed by replacing the split with the following:
```python
ip_port.split(":") 
# Fixed:
ip_port.rsplit(":", 1)
```
https://docs.python.org/3/library/stdtypes.html#str.rsplit

This will start from the right and split the string into only two parts on the first occurrence of `:` from the right.

Reply:
Now thinking about it, shouldn't it be `host` instead of `ip`?